src/Common/src/Interop/Unix/libcurl/Interop.SafeCurlHandle.cs

```diff
@@ -56,11 +96,175 @@ public static void DisposeAndClearHandle(ref SafeCurlMultiHandle curlHandle)
                 }
             }
 
+            internal void PollFds(List<libc.pollfd> readyFds)
+            {
+                int count;
+                libc.pollfd[] pollFds;
+
+                readyFds.Clear();
+
+                lock (this)
+                {
+                    // TODO: Avoid the allocation when count is in 100s
+                    count = _fdSet.Count + 1;
+                    pollFds = new libc.pollfd[count];
```

Comment:
_fdSet.Count==0 is a valid situation when AddEasyHandle has triggered the PollFunction but the CurlSocketCallback which needs to add the fd is delayed for some reason. So we would like to poll on the special fd in this case